injector/src/main/java/edu/ucr/cs/riple/injector/Printer.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 Nima Karimipour
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.injector;
+
+import com.github.javaparser.Range;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.ImportDeclaration;
+import edu.ucr.cs.riple.injector.modifications.Modification;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Applies the text modification instances to source file. Text modifications are applied according
+ * to the target element {@link javax.lang.model.element.ElementKind}.
+ */
+public class Printer {
+
+  /** Path to source file. */
+  private final Path path;
+  /** Lines of source file. */
+  private final List<String> lines;
+
+  public Printer(Path path) {
+    this.path = path;
+
+    try {
+      lines = Files.readAllLines(path);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Applies the set of modification to source file.
+   *
+   * @param modifications Set of modifications.
+   */
+  public void applyModifications(Set<Modification> modifications) {
+    // Modification are sorted to start from the last position, to make the changes ineffective to
+    // current computed offsets.
+    modifications.stream()
+        .sorted(
+            Comparator.comparingInt((Modification o) -> o.startPosition.line)
+                .thenComparingInt(o -> o.startPosition.column)
+                .reversed())
+        .forEach(modification -> modification.visit(lines));
```

Comment:
This is a great point and we should definitely document this limitation. We do not support addition and deletion within one run of Injector. I rewrote `injector` based on the use case of the `core` module and in that use case we only request additions or deletions per run of injector. This limitation was resolved in core since we have apis with name `injectFixes` or `removeFixes`. I updated them to reflect this constraint even further. And we do support addition of multiple annotations on single field, added a unit test for that.

3eade0d2d1d3054f73c6835b68077c6c563ea264
e8d887ed529e4a9fc366e6c7f7be6cf61efd69eb
0c9ad7fc897f6f0cf0950d57dee34d88a27012b7